crypto/ec/ecx_meth.c

```diff
@@ -32,11 +34,14 @@ typedef enum {
 } ecx_key_op_t;
 
 /* Setup EVP_PKEY using public, private or generation */
-static int ecx_key_op(EVP_PKEY *pkey, const X509_ALGOR *palg,
+static int ecx_key_op(EVP_PKEY *pkey, int id, const X509_ALGOR *palg,
                       const unsigned char *p, int plen, ecx_key_op_t op)
 {
     X25519_KEY *xkey;
 
+    if (id == 0)
+        id = pkey->ameth->pkey_id;
```

Comment:
What sort of cases benefit from "pass 0 to use the pkey_id from the ameth" as opposed to having callers that would otherwise pass 0 do the dereferencing themselves?  (That is, are we introducing the possibility for subtle bugs later on?)